### PR TITLE
Update version switcher to set 2.0 as preferred version

### PR DESCRIPTION
**Pull request by melissawm**
Also adds a 'preferred' keyword to the stable version, which is used to select which pages receive a version warning banner from the PyData Sphinx Theme.

This should be merged after the 2.0 documentation is published, so that "stable" points to the right pages.

### CODE REVIEW

**Comment by melissawm** [comment]
@charris can you merge this one?

File: _static/versions.json [hunk 1]
```diff
@@ -3,11 +3,17 @@
         "name": "dev",
         "version": "devdocs",
         "url": "https://numpy.org/devdocs/"
+    },
+        {
```
Comment (seberg) [inline _static/versions.json:7]:
```suggestion
    {
```

**Comment by seberg** [comment]
Well, I didn't expect a silly whitespace fix to cause merge conflicts!?  Will look into it, or do you just want to force push @melissawm and then we put it in without things?

**Comment by melissawm** [comment]
I'll update 👍

**Comment by melissawm** [comment]
Should be good to go - thanks @seberg

**Comment by seberg** [comment]
Thanks! I guess it might now only be the preferred thing anymore, but that is good anyway.

Is it correct that it shows "Choose version" for the stable one?

**Comment by melissawm** [comment]
Hmm this may be https://github.com/pydata/pydata-sphinx-theme/issues/1829? Let me check

**Comment by melissawm** [comment]
Looking at the console for the stable docs I see 

```
Uncaught ReferenceError: $ is not defined
    <anonymous> https://numpy.org/doc/_static/versionwarning.js:13
```

Not sure if this is interfering with the PyData version switcher. I'll try to debug

**Comment by mattip** [comment]
stackoverflow says the script is missing a jquery reference, something like
```
<script language="JavaScript" type="text/javascript" src="/js/jquery-1.2.6.min.js"></script>
```

And indeed, when I compare the 2.0 directory to the other versions, I see it is missing a `_static/jquery.js`

**Comment by melissawm** [comment]
Not sure if that fixes the switcher but let's try

**Comment by mattip** [comment]
Something else is wrong, I don't see a call to load the script from the HTML. I will try rebuilding the docs.

**Comment by melissawm** [comment]
Let me know if you need help, I'll be around.

**Comment by melissawm** [comment]
Are you rebuilding @mattip or would you like me to do it?

**Pull request merged**